src/KarmaTiers.sol

```diff
@@ -174,31 +124,21 @@ contract KarmaTiers is Ownable {
      * @return tierId The tier id that matches the karma balance
      */
     function getTierIdByKarmaBalance(uint256 karmaBalance) external view returns (uint8) {
-        uint8 bestTierId = 0;
-        uint256 bestMinKarma = 0;
-
-        for (uint8 i = 1; i <= currentTierId; i++) {
-            Tier memory currentTier = tiers[i];
-            if (!currentTier.active) continue;
-
+        for (uint8 i = 0; i < tiers.length; i++) {
             // Check if user meets the minimum requirement for this tier
-            if (karmaBalance >= currentTier.minKarma) {
-                // Only update if this tier has a higher minKarma requirement
-                if (currentTier.minKarma > bestMinKarma) {
-                    bestTierId = i;
-                    bestMinKarma = currentTier.minKarma;
-                }
+            if (karmaBalance < tiers[i].minKarma) {
```

Comment:
Maybe worth adding a comment that we can't run into underflow here, because `karmaBalance = 0 <= tiers[0].minKarma`